### PR TITLE
fix: support for federated client discovery request - FS-140

### PR DESCRIPTION
**Pull request by David-Henner**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/FS-140" title="FS-140" target="_blank">StoryFS-140</a>  [iOS] As the organizer of a conference call I like to place a conference call in a conversation with users who are part of different federated backends
  </td></table>
  <br />
 

#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

Requests to `/conversations/{cnv_domain}/{cnv}/proteus/messages` returned `400` 

### Causes

The param `client_mismatch_strategy` was missing from the request's body.

### Solutions

Use `Proteus_QualifiedNewOtrMessage` instead of `Proteus_NewOtrMessage` to generate the request's data.

### Testing

Expanded tests to verify that the right message is being used

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [x] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: Sources/Request Strategies/Client Message/ClientMessageRequestFactoryTests.swift [hunk 1]
```diff
@@ -145,28 +145,47 @@ extension ClientMessageRequestFactoryTests {
...
+                return XCTFail()
+            }
+
+            let message = try? Proteus_NewOtrMessage(serializedData: data)
+
             // THEN
             XCTAssertNotNil(request)
+            XCTAssertNotNil(message)
             XCTAssertEqual(request?.path, "/conversations/\(conversationID.transportString())/otr/messages")
+            XCTAssertEqual(message, expectedMessage)
```
Comment (johnxnguyen) [inline Sources/Request Strategies/Client Message/ClientMessageRequestFactoryTests.swift:174]: Don't you need to assert the query parameter for the missing clients strategy in the path?

Comment (johnxnguyen) [reply Sources/Request Strategies/Client Message/ClientMessageRequestFactoryTests.swift:174]: Ah I see now that when we shouldn't ignore missing clients, then there is no query parameter.

**Review by johnxnguyen — APPROVED** [review]

**Review by sb88k — APPROVED** [review]

**Comment by David-Henner** [comment]
release patch

**Comment by zenkins** [comment]
Hello, @David-Henner! Merging PR for command **release patch** for PR #**464**. Will create a **patch** release after merged.
Next version is 260.0.1.

**Pull request merged**